core/src/main/kotlin/com/tschuchort/compiletesting/AbstractKotlinCompilation.kt

```diff
@@ -262,7 +262,7 @@ abstract class AbstractKotlinCompilation<A : CommonCompilerArguments> internal c
             .mapNotNull { url ->
                 val uri = URI.create(url.toString().removeSuffix("/$resourceName"))
                 when (uri.scheme) {
-                    "jar" -> Paths.get(URI.create(uri.schemeSpecificPart.removeSuffix("!")))
+                    "jar" -> Paths.get(URI.create(encodePath(uri.schemeSpecificPart.removeSuffix("!"))))
```

Comment:
Can you try locally if using `rawSchemeSpecificPart` here prevents the crash? In contrast to `schemeSpecificPart` it doesn't do any decoding.

Reply:
Just tried it, and it's working without any issues with `rawSchemeSpecificPart`.

So should we remove this encoding and proceed with this function `rawSchemeSpecificPart()`? Also, was there any intention behind using `schemeSpecificPart()` earlier?

Reply:
> So should we remove this encoding and proceed with this function?

I would say yes, since it seems to work and is simpler than doing the encoding ourselves. But maybe keep the encoding code somewhere in case it turns out that we still need it. It would be nice if we could have a regression test for this bug but I don't think it would be easy. Do you think there are any other problematic symbols that we could at least test manually?

> was there any intention behind using `schemeSpecificPart()` earlier?

Nope. I just stumbled across `rawSchemeSpecificPart` now because I had to look up what `schemeSpecificPart` does.

Reply:
> Do you think there are any other problematic symbols that we could at least test manually?

Apart from whitespace, I can't imagine the path string (_whether in Windows/Mac/Linux_) includes any special symbol 🤔

> Nope. I just stumbled across rawSchemeSpecificPart now because I had to look up what schemeSpecificPart does.

Looking at docs of `schemeSpecificPart`, it returns same output as `rawSchemeSpecificPart` but decodes the output
![image](https://user-images.githubusercontent.com/19620536/226188189-9558e551-b240-49cf-aadc-e62728a9a276.png)